src/frontend/src/lib/utils/token.utils.ts

```diff
@@ -0,0 +1,47 @@
+import { E8S_PER_ICP } from '$lib/constants/icp.constants';
+import { type Token } from '@dfinity/utils';
+
+/**
+ * Truncates the given amount to 8 decimals.
+ *
+ * This is used to then convert the amount to a string or to a number afterwards.
+ */
+const ulpsToE8s = ({ ulps, decimals }: { ulps: bigint; decimals: number }): bigint => {
```

Comment:
This should not be added to the code base. We are already calculting about based on the token decimals in some place. Please reuse existing code.

Reply:
For example from Ethereum `SendTokenWizard`:

```
parseToken({
					value: `${amount}`,
					unitName: $sendTokenDecimals
				})
```

Reply:
Modified function `getMaxTransactionAmount` to simplify.